src/ui/components/results/verticalresultscomponent.js

```diff
@@ -241,7 +242,24 @@ export default class VerticalResultsComponent extends Component {
   getVerticalURL (data = {}) {
     const verticalConfig = this._verticalsConfig.find(config => config.verticalKey === this.verticalKey) || {};
     const verticalURL = this._config.verticalURL || verticalConfig.url || data.verticalURL || this.verticalKey + '.html';
-    return addParamsToUrl(verticalURL, { query: this.query });
+
+    let tabOrder = getDefaultTabOrder(this._verticalsConfig, getUrlParams());
+    let dataTabOrder = [];
+    if (this.core.globalStorage.getState(StorageKeys.NAVIGATION)) {
+      dataTabOrder = this.core.globalStorage.getState(StorageKeys.NAVIGATION).tabOrder;
+    }
+    // We want to persist the params from the existing URL to the new
+    // URLS we create.
+    if (tabOrder !== undefined && dataTabOrder !== undefined) {
+      tabOrder = mergeTabOrder(dataTabOrder, tabOrder, this._verticalsConfig);
+    }
+
+    const params = { query: this.query };
+    const urlParams = getUrlParams();
+    for (const paramKey in params) {
```

Comment:
This can be streamlined a bit. I don't think we need the loop here since params is an object with a single attribute. Can we just do:
urlParams.set('query', this.query) directly?